src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -113,16 +115,25 @@ private function processCacheConfiguration(array $config, ContainerBuilder $cont
     private function processClientsConfiguration(array $config, ContainerBuilder $container, $debug)
     {
         foreach ($config['clients'] as $name => $options) {
-            $client = new Definition($options['class']);
+            if ($parent = $options['extends']) {
+                $client = new DefinitionDecorator($this->getClientServiceId($parent));
```

Comment:
Do we actually need to extend the parent service? Couldn't we handle all this exclusively in the compiler pass by simply adding a DI tag on the original service?